Vizzuality/heco-invest
# Enable access restriction when environment variables are set

This PR adds changes to restrict the access to the front-end application when the `HTTP_AUTH_USERNAME`, `HTTP_AUTH_PASSWORD`, `NEXTAUTH_SECRET` and `NEXTAUTH_URL` environment variables are defined.

The PR also fixes some linting errors and warnings (see first commit).

## Testing instructions

Set these environment variables to run the test:
```
HTTP_AUTH_USERNAME=vizzuality
HTTP_AUTH_PASSWORD=vizzuality
NEXTAUTH_SECRET=magic-cat
NEXTAUTH_URL=http://localhost:3000
```

1. Run the application in **development** mode, with some empty environment variables

You should be able to browse the application straight away.

2. Run the application in **development** mode, with values for each of the environment variables

You must be requested a user and password to access any page.

3. Run the application in **production** mode, with some empty environment variables

You should be able to browse the application straight away.

3. Run the application in **production** mode, with values for each of the environment variables

You must be requested a user and password to access any page.

## Tracking

[LET-1276](https://vizzuality.atlassian.net/browse/LET-1276)
